ansible_roles/roles/azure_create/tasks/main.yml

```diff
@@ -90,7 +90,14 @@
   shell: "cat {{ working_dir }}/add_vars_tf >> {{ working_dir }}/tf//vars.tf"
 
 - name: Add tag info in.
-  shell: "cat {{ working_dir }}/add_main_vars.tf >> {{ working_dir }}/tf/main.tf"
+  shell: "cp {{ working_dir }}/add_main_vars.tf {{ working_dir }}/temp_file.tf"
+
+- name: Add main info in.
+  shell: "cat {{ working_dir }}/tf/main.tf >> {{ working_dir }}/temp_file.tf"
+
+- name: Move file
+  shell: "cp {{ working_dir }}/temp_file.tf {{ working_dir }}/tf/main.tf"
+
```

Comment:
You could `cat {{ working_dir }}/add_main_vars.tf {{ working_dir }}/tf/main.tf` to acheive something similar.  Overwriting the same filename seems a bit hacky

Reply:
have changed it so we first move main.tf, then do a cat of the two files (in appropriate order) to tf/main.tf.

Reply:
Renaming main.tf would impact AWS and GCP cloud pieces also, thus the moving things around.
// has been corrected.